### ISSUE TITLE
Optional ability to block user account after many failed login attempts

### ISSUE THREAD

**Issue by LukasReschke** [nextcloud/server#493]
After many failed login attempts it may be useful to block user accounts completely. This should probably be used in combination to https://github.com/nextcloud/server/issues/492 so IPs that logged-in successfully in the past can also continue to login in the future.

Also we should send out a mail with a link that allows a user to login and bypass the limitation by trusting their IP by doing so.

**Comment by oparoz** [comment]
Not sure about trusting IPs, but blocking and mailing the user is a good idea.

**Comment by tflidd** [comment]
What about browser authentication with ssl-certs (like startssl does)? Not sure if there is an easy setup method (generation, download & install from web-interface) but such machines could have a higher trust level than other clients.

**Comment by Spacefish** [comment]
I think we shouldn't over engineer that feature. Just set a reasonable
retry rate. Like 20 attempts are no problem, than exponential back off,
starting with 5 seconds or so.

Automated dictionary attacks are mitti gated by this, but it won't affect a
user which can't remember the exact spelling of his password and tries some.

Am 02.08.2016 17:08 schrieb "tflidd" <notifications@github.com>:

**Comment by benediktg** [comment]
How about providing a configuration for fail2ban in the documentation?

**Comment by RaspLuca** [comment]
Here I have a configuration of fail2ban (working!):

cd /etc/fail2ban
nano `fail2ban.local

Include a new section in the configuration as follow:

`[nextcloud]

filter = nextcloud
port = http,https
maxretry = X
logpath = /[YourNextCloudDataLocation]/nextcloud.log
enabled = true`

After that create a new filter

cd filter.d
nano nextcloud.conf

Write as follow:

`[INCLUDES]

before = common.conf

[Definition]

_deamon = nextcloud

failregex={"reqId":".*","remoteAddr":".*","app":"core","message":"Login failed: '.*' \(Remote IP: '<HOST>'\)","level":2,"time":".*"}

ignoreregex =`

DONE

Found at http://linuxcoffee.eu/owncloud-ultimate.php

**Comment by Hudaldadi** [comment]
What about a checkbox in the Admin Area?

**Comment by tgoeg** [comment]
+1 for the fail2ban solution.
It is the better solution if your instance gets hammered with login requests, as the webserver+PHP+DB don't have to process every request but they get blocked on the OS layer already.

I'd like to provide a correct fail2ban filter, however the current audit log format does not allow this anymore!

The only thing a bad login causes is the following line (on NC21):
```
{"reqId":"LzrIc1UXGIE4iWIfv625","level":1,"time":"2022-02-14T15:10:26+00:00","remoteAddr":"1.2.3.4","user":"--","app":"admin_audit","method":"POST","url":"/login","message":"Login attempt: \"bad user\"","userAgent":"myAgent"}
```
This is no different than a successful attempt, so it is useless for fail2ban.
Please bring back a log entry that clearly states there was a *failed* attempt.

(on a sidenote: There is a built in brute force protection in Nextcloud right now. This ticket is old. Still, it might not be enough to save you from DoS-style scenarios or high traffic brute force attempts)

**Comment by MrRinkana** [comment]
This feature exists in recent Nextcloud versions (minus the email to restore account). I do not remember when it was introduced.

In admin options > security > Password policy one can specify an amount of failed logins before account is inactivated.
![bild](https://github.com/nextcloud/server/assets/39345772/268c81a6-b6dc-45bd-bbd1-45952cc71f61)

**Comment by joshtrichards** [comment]
Fixed in nextcloud/password_policy#102
